pyquil/api/_quantum_computer.py

```diff
@@ -128,6 +130,115 @@ def run(self, executable: ExecutableDesignator,
+               ``QuantumComputer`` object. If so, skip to step 2. Otherwise, do the following:
+
+                a. Generate a parameterized program corresponding to the ``TomographyExperiment``
+                   (see the ``TomographyExperiment.generate_experiment_program()`` method for more
+                   details on how it changes the main body program to support state preparation,
+                   measurement, and symmetrization).
+                b. Compile the parameterized program into a parametric (binary) executable, which
+                   contains declared variables that can be assigned at runtime.
+
+            2. For each ``ExperimentSetting`` in the ``TomographyExperiment``, we repeat the
+               following:
+
+                a. Build a collection of memory maps that correspond to the various state
+                   preparation, measurement, and symmetrization specifications.
+                b. Run the parametric executable on the QVM or QPU backend, providing the memory map
+                   to assign variables at runtime.
+                c. Extract the desired statistics from the classified bitstrings that are produced
+                   by the QVM or QPU backend, and package them in an ``ExperimentResult`` object.
+
+            3. Return the list of ``ExperimentResults``.
+
+        This method is extremely useful shorthand for running near-term applications and algorithms,
+        which often have this ansatz + settings structure.
+
+        :param experiment: The ``TomographyExperiment`` to run.
+        :param memory_map: A dictionary mapping declared variables / parameters to their values.
+            The values are a list of floats or integers. Each float or integer corresponds to
+            a particular classical memory register. The memory map provided to the ``experiment``
+            method corresponds to variables in the main body program that we would like to change
+            at runtime (e.g. the variational parameters provided to the ansatz of the variational
+            quantum eigensolver).
+        :return: A list of ``ExperimentResult`` objects containing the statistics gathered
+            according to the specifications of the ``TomographyExperiment``.
+        """
+        executable = self.qam._executable
+        # if this experiment was the last experiment run on this QuantumComputer,
+        # then use the executable that is already loaded into the object
+        if self.qam._experiment != experiment:
+            experiment_program = experiment.generate_experiment_program()
+            executable = self.compile(experiment_program)
+            self.qam._experiment = experiment
+
+        if memory_map is None:
+            memory_map = {}
+
+        results = []
+        for settings in experiment:
+            # TODO: add support for grouped ExperimentSettings
+            setting = settings[0]
```

Comment:
Should we warn or raise an exception here if settings is not a tuple of length 1?